### PR TITLE
MCR-3052 Improvements of MCRTransactionableCallable and MCRFixedUserCallable

### PR DESCRIPTION
**Pull request by toKrause**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3052).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/util/concurrent/MCRTransactionableCallable.java [hunk 1]
```diff
@@ -29,80 +29,139 @@
...
     public Callable<V> get() {
         return callable;
     }
 
+
+    protected enum SessionType {
+
+        PROVIDED,
+
+        TREAD_BOUND,
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/util/concurrent/MCRTransactionableCallable.java]: typo THREAD

File: mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserRunnable.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     * with <code>null</code> as the session parameter.
+     */
+    public MCRFixedUserRunnable(Runnable runnable, MCRUserInformation userInfo) {
+        this(runnable, null, userInfo);
+    }
+
+    /**
+     * Creates a new {@link Runnable} encapsulating the {@link #run()} method with a new
+     * a database transaction. The transaction will be created in the context of a session
+     * ant the privileges of the given user information.
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserRunnable.java]: typo ant

File: mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserFailableRunnable.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Override
+    public void run() {
+        try {
+            new MCRFixedUserCallable<>(() -> {
+                runnable.run();
+                return null;
+            }, userInfo).call();
+        } catch (Exception e) {
+            throw new MCRException("Tailed to run nested runnable wit a fixed user", e);
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserFailableRunnable.java]: typo: Tailed + wit
Same in MCRFixedUserRunnable

File: mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserCallable.java [hunk 4]
```diff
@@ -21,55 +21,40 @@
...
-import org.mycore.common.MCRException;
 import org.mycore.common.MCRSession;
-import org.mycore.common.MCRSessionMgr;
+import org.mycore.common.MCRUsageException;
 import org.mycore.common.MCRUserInformation;
 
 /**
  * Encapsulates a {@link Callable} with a mycore session belonging to a specific user and a database transaction.
- * 
- * @author Matthias Eichner
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserCallable.java:32]: Why remove author tags? Think you should rather add yourself there too.

Comment (toKrause) [reply mycore-base/src/main/java/org/mycore/util/concurrent/MCRFixedUserCallable.java:32]: Git can keep track of authorship way better then the authors themselves. The author tags are mostly wrong throughout the entire project and therefore useless.

**Comment by Mewel** [comment]
I would approve this pull request. Just one more idea, because we had the discussion on the mycore call, about the missuse of callable and runnable, it would be nice to have something to support this.

So code like this:

```
new MCRTransactionableRunnable(() -> {
    dao.getActions().forEach(this::resetJobsWithAction);
    LOGGER.info("MCRJob checking is done");
}).run();
```
could be replaced.

Not sure if functional interfaces would be the right choice, and we need classes for that?

**Comment by toKrause** [comment]
I think a complete redesign or an alternative API should be separat PR.

**Review by Mewel — APPROVED** [review]

**Pull request merged**